src/api/resources/sync/pull.py

```diff
@@ -20,44 +39,75 @@ def _get_records(model_class, filters):
             revision.revision_num as revision_revision_num,
             revision.deleted as revision_deleted
         FROM
-            revision
-        LEFT JOIN
             model_select
+        INNER JOIN
+            revision
         ON
-            revision.record_id = model_select."{pk_field_name}"
+            revision.record_id = model_select."__pk__"
         WHERE
-            {" AND ".join(filters)}
+            {" AND ".join(updates_filters)}
         ORDER BY
             revision.updated_on DESC, revision.revision_num  DESC
     """
 
-    return model_class.objects.raw(sql)
+    updates = model_class.objects.raw(updates_sql)
```

Comment:
Any reason not to turn this into a list of dicts, to be consistent with `deletes`?

Reply:
The model instances get serialized with it's Serializer and we don't have that option for deletes.  I think some of this weill be refactored in M1561 so we can speed up the fishspecies call